misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableToContainInAnyOrderNotOrAtMostValuesExpectationsTest.kt

```diff
@@ -0,0 +1,168 @@
+package ch.tutteli.atrium.specs.integration
+
+import ch.tutteli.atrium.api.fluent.en_GB.*
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.*
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.anElementWhichEquals
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.atMostDescr
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.numberOfSuchElements
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSeven
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.separator
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.toContainInAnyOrder
+import ch.tutteli.atrium.testfactories.TestFactory
+
+@Suppress("FunctionName")
+abstract class AbstractIterableToContainInAnyOrderNotOrAtMostValuesExpectationsTest(
+    private val notToContainOrAtMostPairSpec: Pair<(String, String) -> String, Fun3<Iterable<Double>, Int, Double, Array<out Double>>>,
+    private val notToContainPair: Pair<String, (Int) -> String>
+) : ExpectationFunctionBaseTest() {
+
+    val notToContainOrAtMostSpec = notToContainOrAtMostPairSpec.second
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        notToContainOrAtMostSpec.forSubjectLessTest(2, 2.3, arrayOf())
+    )
+
+    fun Expect<Iterable<Double>>.notToContainOrAtMostFun(atLeast: Int, a: Double, vararg aX: Double) =
+        notToContainOrAtMostSpec(this, atLeast, a, aX.toTypedArray())
+
+    val notToContain = notToContainPair.first
+    val errorMsgContainsNot = notToContainPair.second
+
+    @TestFactory
+    fun test_throws_an_illegalArgumentException() = testFactory(notToContainOrAtMostSpec) {
+        it ("for not at all or at most -1 -- only positive numbers") {
+            expect {
+                expect(oneToSeven()).notToContainOrAtMostFun(-1, 0.1)
+            }.toThrow<IllegalArgumentException> { messageToContain("positive number", -1) }
+        }
+        it ("for not at all or at most 0 -- points to $notToContain") {
+            expect {
+                expect(oneToSeven()).notToContainOrAtMostFun(0, 0.1)
+            }.toThrow<IllegalArgumentException> { message { toEqual(errorMsgContainsNot(0)) } }
+        }
+    }
+
+    @TestFactory
+    fun test_happy_case() = testFactory(notToContainOrAtMostSpec) {
+        it("${notToContainOrAtMostPairSpec.first("1.1", "once")} does not throw") {
+            ch.tutteli.atrium.api.verbs.internal.expect(oneToSeven()).notToContainOrAtMostFun(1, 1.1)
```

Comment:
make sure you use the `expect` from the context and not the top-level one (also in other places)
```suggestion
            expect(oneToSeven()).notToContainOrAtMostFun(1, 1.1)
```